ui/src/timeMachine/components/TimeMachineFluxEditor.tsx

```diff
@@ -55,27 +61,29 @@ class TimeMachineFluxEditor extends PureComponent<Props, State> {
         size: 0.75,
         handleDisplay: HANDLE_NONE,
         render: () => {
-          if (ENABLE_MONACO) {
-            const FluxMonacoEditor = require('src/shared/components/FluxMonacoEditor')
-              .default
-            return (
-              <FluxMonacoEditor
-                script={activeQueryText}
-                onChangeScript={onSetActiveQueryText}
-                onSubmitScript={onSubmitQueries}
-                onCursorChange={this.handleCursorPosition}
-              />
-            )
-          }
           return (
-            <FluxEditor
-              script={activeQueryText}
-              status={{type: '', text: ''}}
-              onChangeScript={onSetActiveQueryText}
-              onSubmitScript={onSubmitQueries}
-              suggestions={[]}
-              onCursorChange={this.handleCursorPosition}
-            />
+            <>
+              <Suspense fallback={spinner}>
+                <FeatureFlag name="monacoEditor">
+                  <FluxMonacoEditor
+                    script={activeQueryText}
+                    onChangeScript={onSetActiveQueryText}
+                    onSubmitScript={onSubmitQueries}
+                    onCursorChange={this.handleCursorPosition}
+                  />
+                </FeatureFlag>
+                <NegativeFeatureFlag name="monacoEditor">
```

Comment:
are you suggesting we remove the FeatureFlag component as well?

Reply:
I don't have a strong opinion on this, but perhaps there's a way to extend the existing `FeatureFlag` so that it supports inverse logic?

```
enum FeatureFlagMode {
  Show = 'show',
  Hide = 'show',
}
```
```
<FeatureFlag name="monacoEditor" mode={FeatureFlagMode.Hide}>
  Old editor
</FeatureFlag>
<FeatureFlag name="monacoEditor" mode={FeatureFlagMode.Show}>
  Monaco editor
</FeatureFlag>
```

Reply:
@drdelambre the `FeatureFlag` component is already used in a lot of places as a convenience, so I don't think so. What I'm strongly against is adding our own functionality that returns the inverse or negative of an expression, whether it's a component or a function.

For a regular function example: this function would be a bit confounding to see in code
```typescript
const notEnabled(featureFlag: string): boolean {
   return !isEnabled(featureFlag)
}

if (notEnabled('foo')) {
  console.log('not enabled')
}
```

I think most people would just expect something like this:
```typescript
if (!isEnabled('foo') {
   console.log('not enabled')
}
```

Reply:
what if we take something similar to @alexpaxton was talking about, but instead of `mode`, add something more flexible like `equals`. currently, flags can only be a boolean, and it seems that any more complexity than that should be a trigger for spiking on a new solution, so it shouldn't spiral too far out of control, but that'll carry us through a lot of basic use cases in case we forget to replace it.

Reply:
@drdelambre i think that's much better than having a negative component. the things that stick out to me as risks for that approach, when weighed against using a built-in `if/else` construct, is that the control flow is removed from a single place (the `if` expression) and spread into two places (the `equals` props of the components). it's spread to two places for boolean expressions, but it could be a large number of places for an enum, string, or number.

```tsx
{isEnabled('monaco') ?
  <MonacoEditor />
:
  <OldEditor />
}
```
vs

```tsx
const monacoIsEnabled = isEnabled('monaco')

<FeatureFlag equals={monacoIsEnabled}>
  <MonacoEditor />
</FeatureFlag>
<FeatureFlag equals={!monacoIsEnabled}>
  <OldEditor />
</FeatureFlag>
```

This spreads a single logical expression `if (isEnabled('monaco'))` `render Monaco` `else` `render Old` into two places: `equals={monacoIsEnabled}` and `equals={!monacoIsEnabled}`. i think that will lead to code that is just more likely to break and slightly more brittle, but if we really really want to do the FeatureFlag component way, and we're okay with the risks and possibly having to be more diligent in PRs to make sure our logic is sound, then go for it :+1:

Reply:
I pushed up a potential solution that feels like a happy medium. I'm not disagreeing with you about it not being the best solution, I just dont feel like this should be the trigger for a spike on replacing the feature flag pattern we currently have and risk fragmentation within the code base. There's a lot of other things i'd love to do with feature flags (centralized / api driven, user groups, testing segments, reporting, etc), but that'll probably be swept up as a prereq when we drop in amplitude or something.